### PR TITLE
Added splitfasta module

### PR DESCRIPTION
**Pull request by weaglesBio**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/selfcomp/splitfasta/main.nf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    path("*.agp")                , emit: agp
+    path "versions.yml"          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+    
+    script:
+    def args = task.ext.args ?: ''
+    """
+    split_genomes_for_ensembl.pl $fasta ${meta.id}_split.fa ${meta.id}_split.agp
```
Comment (priyanka-surana) [inline modules/selfcomp/splitfasta/main.nf:26]: Define prefix: `def prefix = task.ext.prefix ?: "${meta.id}"`
Example usage: https://github.com/nf-core/modules/blob/master/modules/samtools/view/main.nf

**Review by priyanka-surana — CHANGES_REQUESTED** [review]
- Missing modified `tests/config/pytest_modules.yml`
- Run linting
- Make sure tests are run successfully and add details to PR

**Comment by weaglesBio** [comment]
Testing performed:
- Input used: >60Mb .fasta format file (pEimTen1.contigs.fasta)
- Commands:
    - PROFILE=singularity nextflow run tests/modules/selfcomp/splitfasta -entry test_selfcomp_splitfasta -c tests/config/nextflow.config
    - NF_CORE_MODULES_TEST=1 TMPDIR=~ PROFILE=singularity pytest --tag selfcomp/splitfasta --symlink --keep-workflow-wd

**Review by priyanka-surana — APPROVED** [review]
Since everything runs well. I am approving it for merging.

**Pull request merged**